libethereum/ExtVM.cpp

```diff
@@ -134,3 +134,26 @@ void ExtVM::suicide(Address _a)
 	m_s.subBalance(myAddress, m_s.balance(myAddress));
 	ExtVMFace::suicide(_a);
 }
+
+h256 ExtVM::blockHash(u256 _number)
+{
+	u256 const currentNumber = envInfo().number();
+
+	if (_number >= currentNumber || _number < (std::max<u256>(256, currentNumber) - 256))
+		return h256();
+
+	if (currentNumber < m_sealEngine.chainParams().u256Param("metropolisForkBlock") + 256)
+	{
+		assert(envInfo().lastHashes().size() > (unsigned)(currentNumber - 1 - _number));
```

Comment:
The assertion seems to fail if `currentNumber == 0` and `_number` == `0xff..ff` (which is `0 - 1`).  I think `lastHashes().size()` should be zero in that case, and the right hand side evaluates to zero.